### PR TITLE
Feature LWDEV-4338 maintenance mode on frontend

### PR DESCRIPTION
**Pull request by andrey-anokhin**

### CODE REVIEW

File: src/MarginTrading.Frontend/Controllers/IsAliveController.cs [hunk 1]
```diff
@@ -37,29 +37,29 @@ public class IsAliveController : Controller
...
+            return result;
+        }
+
+        private async Task<string> GetBackendVersion(bool isLive)
+        {
             try
             {
-                var responce = await _httpRequestService.GetAsync<IsAliveResponse>("isAlive");
-                result.LiveVersion = responce.Version;
+                var responce = await _httpRequestService.GetAsync<IsAliveResponse>("isAlive", isLive, 3);
```
Comment (GSailen) [inline src/MarginTrading.Frontend/Controllers/IsAliveController.cs]: Note: better to move constants to static resource

File: src/MarginTrading.Frontend/Repositories/Contract/IMaintenanceInfo.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+
+namespace MarginTrading.Frontend.Repositories.Contract
+{
+    public interface IMaintenanceInfo
```
Comment (GSailen) [inline src/MarginTrading.Frontend/Repositories/Contract/IMaintenanceInfo.cs]: Better to split interfaces and implementation to separate files.
And it's not a Repository actually, it's more like an entity, so better to move it to Entities folder
Found in multiple places.

File: src/MarginTrading.Frontend/Services/HttpRequestService.cs [hunk 3]
```diff
@@ -71,61 +60,72 @@ Task<TResponse> Request(bool isLive, bool isTradingEnabled)
                     () => flurlClient.PostJsonAsync(request).ReceiveJson<TResponse>(),
                     ex => ex is FlurlHttpException && !new int?[] {400, 500}.Contains((int?) ((FlurlHttpException) ex).Call.HttpStatus),
                     6,
```
Comment (GSailen) [inline src/MarginTrading.Frontend/Services/HttpRequestService.cs]: Better to isolate constants via statis class of settings

File: src/MarginTrading.Frontend/Services/HttpRequestService.cs [hunk 4]
```diff
@@ -71,61 +60,72 @@ Task<TResponse> Request(bool isLive, bool isTradingEnabled)
...
             catch (Exception ex)
             {
-                ProcessException(isLive, action, request.ToJson(), ex, false);
-                throw;
+                throw new Exception(GetErrorMessage(isLive, action, request.ToJson(), ex));
             }
         }
 
-        public async Task<TResponse> GetAsync<TResponse>(string path, bool isLive = true)
+        public async Task<TResponse> GetAsync<TResponse>(string path, bool isLive = true, int timeout = 30)
```
Comment (GSailen) [inline src/MarginTrading.Frontend/Services/HttpRequestService.cs]: Same here

File: src/MarginTrading.Frontend/Services/HttpRequestService.cs [hunk 5]
```diff
@@ -71,61 +60,72 @@ Task<TResponse> Request(bool isLive, bool isTradingEnabled)
...
         }
+
+        private async Task<IMaintenanceInfo> GetMaintenance(bool isLive)
+        {
+            var cacheKey = CacheKeyBuilder.Create(nameof(HttpRequestService), nameof(CheckMaintenance), isLive);
+
+            return await _cacheProvider.GetAsync(cacheKey,
+                async () => new CachableResult<IMaintenanceInfo>(
+                    await _maintenanceInfoRepository.GetMaintenanceInfo(isLive),
+                    CachingParameters.FromMinutes(1)));
```
Comment (GSailen) [inline src/MarginTrading.Frontend/Services/HttpRequestService.cs]: Same here

**Comment by andrey-anokhin** [comment]
Comments about route constants and separate files are tech debts. 
Will be refactored sometime in future  in the whole solution.

**Pull request merged**